crates/clairvoyance/src/uniswap.rs

```diff
@@ -15,68 +15,129 @@ use bindings::uniswap_v3_factory::UniswapV3Factory;
 use eyre::Result;
 use std::error::Error;
 
+const FACTORY: &str = "0x1F98431c8aD98523631AE4a59f267346ea31F984";
 /// Representation of a pool.
 #[derive(Debug, Clone)]
 pub struct Pool {
     /// Token 0.
-    pub token_0: Token,
+    token_0: Token,
     /// Token 1.
-    pub token_1: Token,
+    token_1: Token,
+    /// Basis point of pool.
+    bp: u32,
     /// Address of the pool.
-    pub address: H160,
+    address: H160,
     /// Factory that created the pool. This could be generic in future.
-    pub factory: UniswapV3Factory<Provider<Http>>,
+    factory: UniswapV3Factory<Provider<Http>>,
     /// Pool contract object.
-    pub inner: IUniswapV3Pool<Provider<Http>>,
+    inner: IUniswapV3Pool<Provider<Http>>,
+    /// Current Tick.
+    tick: i32,
+    /// Current liquidity.
+    liquidity: u128,
+    /// sqrt_price_x96
+    sqrt_price_x96: ethers::types::U256,
 }
 
 impl Pool {
+    /// Getters and setters for error checking and access controls.
```

Comment:
no point in writing getter and setter functions when we can just make the struct field public...

Reply:
yeah but we don't want to make them public because we want to ensure they are always correct and thus only changed by functions with the correct logic.

Reply:
How bout we use this crate? https://github.com/dtolnay/readonly Apologies for my confusion.